### DISCUSSION TITLE
How to fit a model to experimental data

### DISCUSSION THREAD

**Discussion by manishkothakonda** [pybamm-team/PyBaMM#2984] [General]
I am attaching a screenshot of the data I am working on here, which is constant discharge data of a 10Ah battery from 100% SOC to 0% SOC for a 0.1C discharge rate for a temperature of 40°C

I have other data in the spreadsheet related to 10Ah battery from 100% SOC to 0% SOC for a  0.2C, and  1C discharge rate for a temperature of 40°C,  25°C,  5°C,  and -25°C,

I would like some assistance on what path could I take to fit this data into a model, and the parameters to optimize.
I would like to evaluate the fitted battery model to 1C constant current discharge from 100% SOC at an ambient temperature of -25°C for the following time periods in seconds: t = 0, 20, 40, 60, 80, 100, …, 3000, 3020, 3040, 3060, 3080, 3100

This is my first time using PyBaMM and I am an adept user of Python.

<img width="246" alt="Screen Shot 2023-05-24 at 10 26 03 AM" src="https://github.com/pybamm-team/PyBaMM/assets/32824488/f0bd32d1-2193-477d-830c-268dc9bd785f">

**Comment by brosaplanella** [comment]
You might want to try [pybamm-param](https://github.com/paramm-team/pybamm-param), which is a package for fitting PyBaMM models to data. Note that the package is under development so is neither stable nor well-tested at the moment.

We can not advise on which parameters to fit, as this is a research question.

**Reply by manishkothakonda** [reply]
Thanks, @brosaplanella, I appreciate you pointing out the package. However, I am unable to install the pybamm-param package. I tried installing in mac os following the exact commands. The error I see is when I do "pip install -e .[dev]", which replies back as "zsh: no matches found: .[dev]" 


Are there specific guidelines that you could share to help me install this python package?

Thank you very much

**Reply by brosaplanella** [reply]
Try installing directly from Github: `pip install git+https://github.com/paramm-team/pybamm-param`

**Reply by manishkothakonda** [reply]
Thanks, This works!

**Reply by agriyakhetarpal** [reply]
Hi, the pattern matching for `zsh`-based shells is different from that of `bash` – you should use `pip install -e .'[dev]'` instead (enclosing it with single quotes) and it should work :)
